docroot/modules/custom/va_gov_links/src/EventSubscriber/ContentReleaseBrokenLinksIngestion.php

```
@@ -0,0 +1,126 @@
+  /**
+   * Logger.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected $logger;
+  /**
+   * Settings Service.
+   *
+   * @var \Drupal\Core\Site\Settings
+   */
+  protected $settings;
+
+  /**
+   * Constructor for ContentReleaseBrokenLinksSubscriber objects.
+   *
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   Http client.
+   * @param \Drupal\Core\State\StateInterface $state
+   *   The state service.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger
+   *   Logger logger logger logger.
+   * @param \Drupal\Core\Site\Settings $settings
+   *   Settings.
+   */
+  public function __construct(ClientInterface $http_client, StateInterface $state, LoggerChannelFactoryInterface $logger, Settings $settings) {
+    $this->httpClient = $http_client;
+    $this->state = $state;
+    $this->logger = $logger;
+    $this->settings = $settings;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function getSubscribedEvents() {
+    $events[ReleaseStateTransitionEvent::NAME] = 'retrieveBrokenLinks';
+    return $events;
+  }
+
+  /**
+   * Retrieve the broken links report when a content release finishes.
+   *
+   * @param \Drupal\va_gov_build_trigger\Event\ReleaseStateTransitionEvent $event
+   *   The release state transition event.
+   */
+  public function retrieveBrokenLinks(ReleaseStateTransitionEvent $event) {
+    if ($this->settings->get('broken_link_report_import_enabled')) {
+      if ($event->getNewReleaseState() === ReleaseStateManager::STATE_READY) {
+        if (in_array($event->getOldReleaseState(), [
+          ReleaseStateManager::STATE_COMPLETE,
+          ReleaseStateManager::STATE_ERROR,
+        ])) {
+          $report_location = $this->settings->get('broken_link_report_location');
+          // If the report location starts with http, use an http client to
+          // retrieve it. Otherwise, assume it is a local file and attempt to
+          // read it.
```

Changes:
* This is now two try/catch blocks, so that we can log different error messages.
* No longer `throw`ing in the catch block, since that still results in an uncaught exception.
* Using logger for all messaging, including errors.

---

Would an uncaught exception pose a problem?  It might be appropriate in this case -- Sentry should post a comment in #cms-notifications, I'd get tagged, and would know, oh, something's up with the thing.  And I don't mind that.  Better that than silent failure, but that might be better with #10813.

But that's a matter of taste and comfort.

If this code's split out, we can reconsider that then, too.

---

It's throwing an error in the CMS, so I would expect Sentry to report that. If it doesn't, we should probably amend that.

What changes is, the queue run will continue; the failure won't halt subsequent code. 

I agree we don't want it to fail silently.